### PR TITLE
[TreeView] Fix `alpha` usage with  CSS variables

### LINKED ISSUE #14349: [tree view] TreeItem2DragAndDropOverlay - Error: MUI: Unsupported `var(--palette-primary-dark)` color.
### Steps to reproduce

use `<CssVarsProvider/>` for base example

https://mui.com/x/react-tree-view/rich-tree-view/ordering/#only-trigger-the-reordering-from-a-drag-handle

### Current behavior

xx-1

to my knowledge the `alpha` function will not support vars so this could be the cause of the error
xxx

https://github.com/mui/mui-x/blob/master/packages/x-tree-view/src/TreeItem2DragAndDropOverlay/TreeItem2DragAndDropOverlay.tsx

### Expected behavior

No error with`TreeItem2DragAndDropOverlay`

### Context

_No response_

### Your environment

_No response_

**Search keywords**: TreeItem2DragAndDropOverlay

### PR DESCRIPTION
**Pull request by wangkailang**
Fixes #14349.

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/mui-x/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

### CODE REVIEW

File: packages/x-tree-view/src/TreeItem2DragAndDropOverlay/TreeItem2DragAndDropOverlay.tsx [hunk 1]
```diff
@@ -24,26 +24,26 @@ const TreeItem2DragAndDropOverlayRoot = styled('div', {
       style: {
         marginLeft: 'calc(var(--TreeView-indentMultiplier) * var(--TreeView-itemDepth))',
         borderRadius: theme.shape.borderRadius,
-        backgroundColor: alpha((theme.vars || theme).palette.primary.dark, 0.15),
+        backgroundColor: alpha(theme.palette.primary.dark, 0.15),
```
Comment (LukasTy) [inline packages/x-tree-view/src/TreeItem2DragAndDropOverlay/TreeItem2DragAndDropOverlay.tsx:27]: The correct solution can be found here: https://github.com/mui/mui-x/blob/3f12d3de1b6961105493978f48c532375289f7fa/packages/x-tree-view/src/TreeItem2/TreeItem2.tsx#L84-L86
Would you be able to apply the necessary changes? 🤔

**Review by LukasTy — COMMENTED** [review]
Thank you for spotting this issue! 🙏 

I've linked an example with an expected approach to such cases. 😉

File: packages/x-tree-view/src/TreeItem2DragAndDropOverlay/TreeItem2DragAndDropOverlay.tsx (hunk 1 above)
Comment (wangkailang) [reply packages/x-tree-view/src/TreeItem2DragAndDropOverlay/TreeItem2DragAndDropOverlay.tsx:27]: ok

**Comment by wangkailang** [comment]
https://github.com/mui/mui-x/issues/14349

**Comment by flaviendelangle** [comment]
June noticed that in #14772, but nice to have it fixed independently :ok_hand:

File: packages/x-tree-view/src/TreeItem2DragAndDropOverlay/TreeItem2DragAndDropOverlay.tsx [hunk 2]
```diff
@@ -52,9 +58,11 @@ const TreeItem2DragAndDropOverlayRoot = styled('div', {
       style: {
         marginLeft:
           'calc(var(--TreeView-indentMultiplier) * calc(var(--TreeView-itemDepth) - 1))' as any,
-        borderBottom: `1px solid ${alpha((theme.vars || theme).palette.grey[900], 0.6)}`,
+        borderBottom: theme.vars
+          ? `1px solid rgba(${theme.vars.palette.grey[900]} / ${0.6})`
+          : `1px solid ${alpha(theme.palette.grey[900], 0.6)}`,
         ...theme.applyStyles('dark', {
-          borderBottomColor: alpha((theme.vars || theme).palette.grey[900], 0.6),
+          borderBottomColor: theme.vars ? `rgba(${theme.vars.palette.grey[900]} / ${0.6})` : alpha(theme.palette.grey[900], 0.6),
```
Comment (LukasTy) [inline packages/x-tree-view/src/TreeItem2DragAndDropOverlay/TreeItem2DragAndDropOverlay.tsx:65]: @flaviendelangle @noraleonte could you double check if this is not an error? 🤔 
It seems that this should maybe use `grey[100]` as in all other cases for the dark theme. 🤔 
Now it does nothing... 🤷

Comment (LukasTy) [inline packages/x-tree-view/src/TreeItem2DragAndDropOverlay/TreeItem2DragAndDropOverlay.tsx:65]:
```suggestion
          borderBottomColor: theme.vars ? `rgba(${theme.vars.palette.grey[100]} / ${0.6})` : alpha(theme.palette.grey[100], 0.6),
```

**Review by LukasTy — APPROVED** [review]
LGTM, thank you for taking care of it. 👍

**Pull request merged**